ansible_collections/arista/avd/roles/eos_designs/python_modules/l3_edge/utils.py

```diff
@@ -0,0 +1,376 @@
+    """
+
+    # Set type hints for Attributes of the main class as needed
+    _hostvars: dict
+    _avd_ip_addressing: AvdIpAddressing
+    _avd_interface_descriptions: AvdInterfaceDescriptions
+
+    @cached_property
+    def _l3_edge(self) -> list:
+        return get(self._hostvars, "l3_edge")
+
+    @cached_property
+    def _p2p_links_profiles(self) -> list:
+        return convert_dicts(get(self._hostvars, "l3_edge.p2p_links_profiles", default=[]), "name")
+
+    @cached_property
+    def _p2p_links_ip_pools(self) -> list:
+        return convert_dicts(get(self._hostvars, "l3_edge.p2p_links_ip_pools", default=[]), primary_key="name", secondary_key="ipv4_pool")
+
+    @cached_property
+    def _p2p_links(self) -> list:
+        return get(self._hostvars, "l3_edge.p2p_links", default=[])
+
+    @cached_property
+    def _hostname(self) -> str:
+        return get(self._hostvars, "switch.hostname", required=True)
+
+    @cached_property
+    def _p2p_uplinks_mtu(self) -> str:
+        return get(self._hostvars, "p2p_uplinks_mtu", required=True)
+
+    @cached_property
+    def _p2p_uplinks_qos_profile(self) -> str | None:
+        return get(self._hostvars, "p2p_uplinks_qos_profile")
+
+    @cached_property
+    def _underlay_rfc5549(self) -> bool:
+        return get(self._hostvars, "underlay_rfc5549") is True
+
+    @cached_property
+    def _underlay_bgp(self) -> bool:
+        return get(self._hostvars, "switch.underlay.bgp") is True
+
+    @cached_property
+    def _underlay_ospf(self) -> bool:
+        return get(self._hostvars, "switch.underlay.ospf") is True
+
+    @cached_property
+    def _underlay_isis(self) -> bool:
+        return get(self._hostvars, "switch.underlay.isis") is True
+
+    @cached_property
+    def _underlay_ldp(self) -> bool:
+        return get(self._hostvars, "switch.underlay.ldp") is True
+
+    @cached_property
+    def _mpls_lsr(self) -> bool:
+        return get(self._hostvars, "switch.mpls_lsr") is True
+
+    @cached_property
+    def _underlay_ospf_area(self) -> str:
+        return get(self._hostvars, "underlay_ospf_area", required=True)
+
+    @cached_property
+    def _underlay_ospf_process_id(self) -> int:
+        return get(self._hostvars, "underlay_ospf_process_id", required=True)
+
+    @cached_property
+    def _isis_instance_name(self) -> str:
+        return get(self._hostvars, "switch.isis_instance_name", required=True)
+
+    @cached_property
+    def _isis_default_metric(self) -> int | None:
+        return get(self._hostvars, "isis_default_metric")
+
+    @cached_property
+    def _isis_default_circuit_type(self) -> str | None:
+        return get(self._hostvars, "isis_default_circuit_type")
+
+    @cached_property
+    def _peer_group_ipv4_underlay_peers_name(self) -> str:
+        return get(self._hostvars, "switch.bgp_peer_groups.ipv4_underlay_peers.name", required=True)
+
+    @cached_property
+    def _bgp_as(self) -> str | None:
```

Comment:
could need to translate to string here

Reply:
we are reading from switch.* which is our own facts. It should be guaranteed to be a string at this point.